File: forgeboot-trace/src/main/java/com/gewuyou/forgeboot/trace/util/RequestIdUtil.java
package com.gewuyou.forgeboot.trace.util;

import java.util.UUID;

/**
 * 请求 ID Util
 * 这个类需配合 RequestIdFilter 使用，用于生成请求 ID，并将其绑定到线程变量中，供后续可能需要的地方使用。
 * @author gewuyou
 * @since 2025-01-02 14:27:45
 */
public class RequestIdUtil {
    private static final ThreadLocal<String> REQUEST_ID_HOLDER = new ThreadLocal<>();
    private RequestIdUtil() {
    }
    public static void generateRequestId() {
        REQUEST_ID_HOLDER.set(UUID.randomUUID().toString());
    }

    public static String getRequestId() {
        return REQUEST_ID_HOLDER.get();
    }

    public static void setRequestId(String uuid) {
        REQUEST_ID_HOLDER.set(uuid);
    }

    public static void removeRequestId() {
        REQUEST_ID_HOLDER.remove();
    }
}

File: forgeboot-trace/forgeboot-trace-impl/src/main/kotlin/com/gewuyou/forgeboot/trace/impl/util/RequestIdUtil.kt
package com.gewuyou.forgeboot.trace.impl.util

import java.util.*

/**
 * 请求 ID Util
 * 这个类需配合 RequestIdFilter 使用，用于生成请求 ID，并将其绑定到线程变量中，供后续可能需要的地方使用。
 * @author gewuyou
 * @since 2025-01-02 14:27:45
 */
 object RequestIdUtil {
    private val REQUEST_ID_HOLDER = ThreadLocal<String>()
    fun generateRequestId() {
        REQUEST_ID_HOLDER.set(UUID.randomUUID().toString())
    }

    var requestId: String?
        get() = REQUEST_ID_HOLDER.get()
        set(uuid) {
            REQUEST_ID_HOLDER.set(uuid)
        }

    fun removeRequestId() {
        REQUEST_ID_HOLDER.remove()
    }
}
